xarray/core/datatree.py

```diff
@@ -1060,6 +1060,50 @@ def drop_nodes(
         result._replace_node(children=children_to_keep)
         return result
 
+    def move(self, origin: str, destination: str, parents: bool = False) -> DataTree:
+        """
+        Move a node to a different location in the tree.
+
+        Parameters
+        ----------
+        origin: str
+            The node to move.
+        destination: str
+            The new node destination.
+        parents: bool, optional
```

Comment:
I think this argument should be renamed, though I'm not sure what to. Perhaps `create_intermediates`?

Reply:
I am not sure either, I took inspiration from pathlib: https://docs.python.org/3/library/pathlib.html#pathlib.Path.mkdir
Happy to change to anything.

Reply:
Ah I see. Taking inspiration from pathlib was a good idea. But I don't think it's a great name in this case...

The `exists_ok` arg in `pathlib.Path.mkdir` is also potentially relevant - the user might want to avoid replacing the subtree already at that path.

Reply:
Ah, good point. I will add an `exists_ok` argument for now, until we settle on names for these arguments.